File: app/src/test/java/com/farmerbb/taskbar/util/DashboardHelperTest.java
package com.farmerbb.taskbar.util;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class DashboardHelperTest {
    private DashboardHelper helper = DashboardHelper.getInstance();

    @Test
    public void testGetInstance() {
        assertNotNull(helper);
        for (int i = 1; i <= 20; i++) {
            assertEquals(helper, DashboardHelper.getInstance());
        }
    }

    @Test
    public void testSetDashboardOpen() {
        assertFalse(helper.isDashboardOpen());
        helper.setDashboardOpen(true);
        assertTrue(helper.isDashboardOpen());
        helper.setDashboardOpen(false);
    }
}

File: app/src/test/java/com/farmerbb/taskbar/helper/DashboardHelperTest.kt
package com.farmerbb.taskbar.helper

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class DashboardHelperTest {
    private val helper = DashboardHelper.getInstance()

    @Test
    fun testGetInstance() {
        Assert.assertNotNull(helper)
        for (i in 1..20) {
            Assert.assertEquals(helper, DashboardHelper.getInstance())
        }
    }

    @Test
    fun testSetDashboardOpen() {
        Assert.assertFalse(helper.isDashboardOpen)
        helper.isDashboardOpen = true
        Assert.assertTrue(helper.isDashboardOpen)
        helper.isDashboardOpen = false
    }
}
